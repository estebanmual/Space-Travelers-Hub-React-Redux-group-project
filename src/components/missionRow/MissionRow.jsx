import { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Badge } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { joinMission } from '../../Redux/Missions/missions';
import style from './MissionRow.module.css';

function MissionRow({ mission }) {
  const {
    name, description, reserved, id,
  } = mission;

  const [reservedMission, setReservedMission] = useState(reserved);

  const dispatch = useDispatch();

  return (
    <tr className={style.Row}>
      <td className={style.Name}>{name}</td>
      <td className={style.Description}>{description}</td>
      <td className={style.Status}>
        {reservedMission ? <Badge bg="primary">Active Member</Badge> : <Badge bg="secondary">NOT A MEMBER</Badge>}
      </td>
      <td className={style.Button}>
        {reservedMission ? <Button id={id} variant="outline-danger" onClick={(e) => { setReservedMission(false); dispatch(joinMission(e.target.id)); }}>Leave Mission</Button> : <Button id={id} variant="outline-dark" onClick={(e) => { setReservedMission(true); dispatch(joinMission(e.target.id)); }}>Join Mission</Button>}
      </td>
    </tr>
  );
}

export default MissionRow;

MissionRow.defaultProps = {
  mission: {
    id: null,
    name: '',
    description: '',
    reserved: false,
  },
};

MissionRow.propTypes = {
  mission: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    description: PropTypes.string,
    reserved: PropTypes.bool,
  }),
};
