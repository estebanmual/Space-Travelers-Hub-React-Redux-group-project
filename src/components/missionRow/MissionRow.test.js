import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../Redux/storeSetUp';
import MissionRow from './MissionRow';

describe('testing Header', () => {
  it('snapshot for the header ', () => {
    const tree = renderer.create(
      <Provider store={store}><table><tbody><MissionRow /></tbody></table></Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Header component renders correctly', () => {
    render(<Provider store={store}><MissionRow /></Provider>);
    expect(screen.getByText('NOT A MEMBER')).toBeInTheDocument();
  });
  it('click button', () => {
    const button = render(<Provider store={store}><MissionRow /></Provider>).getByText('Join Mission');
    const mockhandler = jest.fn();

    if (fireEvent.click(button)) {
      mockhandler();
    }

    expect(mockhandler).toHaveBeenCalledTimes(1);
  });
  it('click Leave mission', () => {
    const mission = {
      id: '',
      name: '',
      description: '',
      reserved: true,
    };
    const button = render(<Provider store={store}><MissionRow mission={mission} /></Provider>).getByText('Leave Mission');
    const mockhandler = jest.fn();
    if (fireEvent.click(button)) { mockhandler(); }

    expect(mockhandler).toHaveBeenCalledTimes(1);
  });
});
