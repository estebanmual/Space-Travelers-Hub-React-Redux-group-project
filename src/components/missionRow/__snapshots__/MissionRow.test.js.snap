// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header snapshot for the header  1`] = `
<table>
  <tbody>
    <tr
      className="Row"
    >
      <td
        className="Name"
      />
      <td
        className="Description"
      />
      <td
        className="Status"
      >
        <span
          className="badge bg-secondary"
        >
          NOT A MEMBER
        </span>
      </td>
      <td
        className="Button"
      >
        <button
          className="btn btn-outline-dark"
          disabled={false}
          id={null}
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
